geode-management-api/src/main/java/org/apache/geode/cache/configuration/DeclarableType.java

```diff
@@ -155,21 +145,7 @@ public String toString() {
         + parameters.stream().map(Objects::toString).collect(Collectors.joining(",")) + "}";
   }
 
-  public <T> T newInstance(Cache cache) {
-    try {
-      Class<T> loadedClass = (Class<T>) ClassPathLoader.getLatest().forName(className);
-      T object = loadedClass.newInstance();
-      if (object instanceof Declarable) {
-        Declarable declarable = (Declarable) object;
-        Properties initProperties = new Properties();
-        for (ParameterType parameter : parameters) {
-          initProperties.put(parameter.getName(), parameter.newInstance(cache));
-        }
-        declarable.initialize(cache, initProperties);
-      }
-      return object;
-    } catch (Exception e) {
-      throw new RuntimeException("Error instantiating class: <" + className + ">", e);
-    }
+  public <V> V newInstance(ConfigTypeInstantiator<DeclarableType> instantiator) {
```

Comment:
I see the need to move this functionality out of here. We can probably completely get rid of this method, and have an instantiator to instantiate the instance. e.g.
```
mutator.setCacheLoader(ConfigInstantiator.instantiate(config, cache));
```